### PR TITLE
mkcloud: Allow custom crowbar networking mode

### PR DESCRIPTION
**Pull request by toabctl**
Introduce a new 'crowbar_networkingmode' variable which defaults to 'single'.
When set to 'team' ('nics' variable must be > 1 then), mkcloud deploys the cloud
with a bonding network setup. This is useful for testing more complex (and more
realistic) scenarios.
Note: The crowbar node itself has no bonding enabled. Our current mkcloud setup
gives the crowbar node only a single nic (even if 'nics' > 1). That's why this
commit adds an extra entry to the conduit_map for the crowbar node.

It can be used to set the Crowbar networking mode which can be 'team' which
does bond multiple interfaces.
This is useful for testing more realistic scenarios where multiple interfaces
are used together.

### CODE REVIEW

**Comment by toabctl** [comment]
Only commit msg adjusted. All checks passed already

File: jenkins/ci.suse.de/openstack-mkcloud.yaml [hunk 1]
```diff
@@ -160,6 +160,12 @@
           default:
           description: Number of compute nodes
 
+      - string:
+          name: nics
+          default: '1'
+          description: >-
+            Number of network interfaces for the nodes (admin node has always 1 currenty)
```
Comment (s-t-e-v-e-n-k) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: Typo: "currenty" -> "currently"

Comment (toabctl) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: Fixed

**Comment by nvin-krja** [comment]
I remember when I tried bonding with mkcloud, I had to change the pci number somewhere in mkcloud because addition nic interfaces were overlapping with pci ids we had. I hope that is fixed somehow.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1472,6 +1472,25 @@ EOF
         -e "s/ [47]00/ $vlan_sdn/g" \
         $netfile
 
+    # set requested networking mode for Crowbar
+    /opt/dell/bin/json-edit -a attributes.network.mode -v "$crowbar_networkingmode" $netfile
+
+    if [ "$crowbar_networkingmode" == "team" ]; then
+        # special conduit mapping when having a crowbar admin node with a single
+        # nic but want bonding for the rest of the nodes
+        local conmap_element='{"pattern": "team/1/crowbar",
```
Comment (rhafer) [inline scripts/qa_crowbarsetup.sh]: I think the "crowbar" is somewhat redundant. IIRC the role name is used as a pattern and every node will have a role assigned that matches the pattern "crowbar" (think of the node roles, they are called "crowbar_\<nodename\>"). But this is just a minor nitpick.

Comment (toabctl) [inline scripts/qa_crowbarsetup.sh]: @rhafer according to https://www.suse.com/documentation/suse-openstack-cloud-7/singlehtml/book_cloud_deploy/book_cloud_deploy.html#ex.conduits.role , ```crowbar``` should work

Comment (rhafer) [inline scripts/qa_crowbarsetup.sh]: Seems you just found a bug in the deployment guide :(. The code is really using pattern matching there: https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/lib/crowbar/conduit_resolver.rb#L98

**Pull request merged**